packages/site/src/components/PropList.js

```diff
@@ -0,0 +1,205 @@
+  }),
+  propColumnHeader: (props, theme) => ({
+    borderBottom: `2px solid ${theme.color_gray}`,
+    textAlign: 'left'
+  }),
+  propP: {
+    margin: 0
+  },
+  propName: (props, theme) => ({
+    fontFamily: theme.fontFamily_monospace
+  }),
+  propRequired: (props, theme) => ({
+    color: theme.color_background_danger,
+    fontFamily: theme.fontFamily_monospace
+  }),
+  propRow: (props, theme) => ({
+    borderBottom: `1px solid ${theme.color_gray}`
+  }),
+  propTable: {
+    borderCollapse: 'collapse',
+    borserSpacing: 0,
+    width: '100%'
+  },
+  propType: (props, theme) => ({
+    display: 'inline-block',
+    backgroundColor: theme.color_gray,
+    borderRadius: '3px',
+    fontFamily: theme.fontFamily_monospace,
+    padding: theme.measurement_a
+  }),
+  propValue: (props, theme) => ({
+    border: `1px solid ${theme.color_gray}`,
+    height: '100%',
+    resize: 'vertical',
+    width: '100%'
+  }),
+  root: (props, theme) => ({
+    fontFamily: theme.fontFamily_system,
+    margin: `${theme.measurement_d} 0 0 0`
+  })
+};
+
+const CodeValue = createStyledComponent('span', styles.codeValue);
+const PropTable = createStyledComponent('table', styles.propTable);
+const PropColumnHeader = createStyledComponent('th', styles.propColumnHeader);
+const PropCell = createStyledComponent('td', styles.propCell);
+const PropName = createStyledComponent('span', styles.propName);
+const PropP = createStyledComponent('p', styles.propP);
+const PropRequired = createStyledComponent('span', styles.propRequired);
+const PropRow = createStyledComponent('tr', styles.propRow);
+const PropType = createStyledComponent('span', styles.propType);
+const PropValue = createStyledComponent('textarea', styles.propValue);
+const Root = createStyledComponent('div', styles.root);
+
+function DefaultValue({
+  defaultValue,
+  required
+}: {
+  defaultValue?: any,
+  required?: boolean
+}) {
+  return required
+    ? <PropRequired>required</PropRequired>
+    : <CodeValue>
+        {undefined === defaultValue
+          ? 'undefined'
+          : JSON.stringify(defaultValue)}
+      </CodeValue>;
+}
+
+type OnPropChange = (propDescription: Object, newValue: any) => void;
+
+type DocProps = {
+  defaultValue?: any,
+  description?: string,
+  exampleValue?: any,
+  name: string,
+  onPropChange: OnPropChange,
+  type: string,
+  required?: boolean
+};
+
+function onPropValueChange(
+  e: Object,
+  propDescription: Object,
+  callback: OnPropChange
+) {
+  try {
+    const newValue = JSON.parse(e.target.value);
+    callback(propDescription, newValue);
+  } catch (e) {
+    console.warn('invalid JSON string');
+  }
+}
+
+function PropDoc({
+  defaultValue,
+  description,
+  exampleValue,
+  name,
+  onPropChange,
+  required,
+  type
+}: DocProps) {
+  return (
+    <PropRow>
+      <PropCell><PropName>{name}</PropName></PropCell>
+      <PropCell>
+        <DefaultValue defaultValue={defaultValue} required={required} />
```

Comment:
As mentioned previously, I dislike combining the `default` and `required` fields.  Consider the following...

A string prop is required.  It is represented as having a default value of "required", which is likely not the actual default value, though that is what it makes me think.  This issue is also heightened for our visually impaired users as the "required" only differs based on color.  While I can certainly figure it out, I don't understand the desire to join these fields.

I would suggest to consider adding a small required `tag` or an `*` to the `prop` field.

Prior art:
* https://polaris.shopify.com/components/navigation/tabs#navigation
* http://www.material-ui.com/#/components/badge